File: yh_check/scripts/yh_check_sub.py
#!/usr/bin/python3
# _*_ coding: utf-8 _*_

import rospy
from yh_check.msg import YhCheck

class MyCheck:
    def __init__(self):
        self.sub_distance = rospy.Subscriber("check_distance", YhCheck, self.distanceCallback)
        self.sub_camera = rospy.Subscriber("check_camera", YhCheck, self.cameraCallback)
        self.distance = True
        self.camera = True
    
    def distanceCallback(self, msg):
        self.distance = msg.data
        if self.distance and self.camera:
            rospy.loginfo("Ok")

    def cameraCallback(self, msg):
        self.camera = msg.data
        if self.distance and self.camera:
            rospy.loginfo("Ok")

if __name__ == "__main__":
    rospy.init_node("yh_check_sub")
    my_check = MyCheck()
    rospy.spin()
        

File: yh_check/src/yh_check_sub.cpp
#include "ros/ros.h"
#include "yh_check/YhCheck.h"

/*
class MySub{
    public:
        // 함수
        MySub(void){
            sub_distance = nh.subscribe("check_distance", 10, &MySub::distanceCallback, this);
            sub_camera = nh.subscribe("check_camera", 10, &MySub::cameraCallback, this);
        }
    private:
        // 변수
        ros::NodeHandle nh;
        ros::Subscriber sub_distance;
        ros::Subscriber sub_camera;
        bool distance = true;
        bool camera = true;
};
*/

bool distance = true;
bool camera = true;

void distanceCallback(const yh_check::YhCheck::ConstPtr& msg){
    distance = msg->data;
    if(distance && camera){
        ROS_INFO("OK");
    }
}

void cameraCallback(const yh_check::YhCheck::ConstPtr& msg){
    camera = msg->data;
    if(distance && camera){
        ROS_INFO("OK");
    }
}

int main(int argc, char** argv){
    ros::init(argc, argv, "yh_check_sub");
    ros::NodeHandle nh;

    ros::Subscriber sub_distance = nh.subscribe("check_distance", 10, distanceCallback);
    ros::Subscriber sub_camera = nh.subscribe("check_camera", 10, cameraCallback);

    ros::spin();

    return 0;
}